sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -148,6 +149,8 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
 
 object SharedState {
 
+  URL.setURLStreamHandlerFactory(new SparkUrlStreamHandlerFactory())
```

Comment:
You could proxy to other `URLStreamHandlerFactory` when `FsUrlStreamHandlerFactory#createURLStreamHandler` returns null.